src/platforms/esp32/components/avm_builtins/CMakeLists.txt

```diff
@@ -22,17 +22,18 @@ set(AVM_BUILTIN_COMPONENT_SRCS
     "gpio_driver.c"
     "i2c_driver.c"
     "ledc_nif.c"
+    "network_driver.c"
     "nvs_nif.c"
     "rtc_slow_nif.c"
     "socket_driver.c"
     "spi_driver.c"
     "uart_driver.c"
 )
 
-if (IDF_VERSION_MAJOR EQUAL 4)
-    list(APPEND AVM_BUILTIN_COMPONENT_SRCS "network_driver.c")
+if (IDF_VERSION_MAJOR GREATER_EQUAL 5)
+    set(ADDITIONAL_PRIV_REQUIRES "esp_hw_support" "efuse")
```

Comment:
Just to understand, I didn't check here, are esp_hw_support and efuse v5 only components?

Reply:
No, those components do exist in IDF 4.x, but in 5.x the they are necessary for obtaining the default MAC address of a network interface. In 4.x versions the MAC address is obtained by using the "soc" component.